cmd/transcriber/call/transcriber.go

```diff
@@ -41,7 +42,13 @@ type Transcriber struct {
 }
 
 func NewTranscriber(cfg config.CallTranscriberConfig) (*Transcriber, error) {
-	if err := cfg.IsValid(); err != nil {
+	if err := cfg.IsValid(true); err != nil {
+		if apiClient, err2 := CreateAPIClient(cfg); err2 == nil {
+			t := &Transcriber{cfg: cfg, apiClient: apiClient}
+			if err3 := t.ReportJobFailure(fmt.Sprintf("failed to validate config: %s", err.Error())); err3 != nil {
+				slog.Error("failed to report job failure", slog.String("err", err3.Error()))
+			}
+		}
```

Comment:
I agree with the intent but maybe we could simplify a bit. I'd make the URL parsing logic its own util function so that we don't have to parse the whole config in order to create the client which can become the first step. 

Then, once we have the client, we can do anything else and report failure on error, ideally in a `defer` block leveraging a named returned (e.g. `retErr`).

Reply:
Done. I like that it's more vertical code, but I've never been a huge fan of named returns. 🤷